### PR TITLE
Sonarcloud integration

### PR DESCRIPTION
**Pull request by iamianmullins**
Sonarcloud integration with updated Readme to display sonarcloud analysis

### CODE REVIEW

File: sonar-project.properties [hunk 1]
```diff
@@ -0,0 +1,12 @@
+sonar.projectKey=iamianmullins_twincam
+sonar.organization=iamianmullins
+
+# This is the name and version displayed in the SonarCloud UI.
```
Comment (ericcurtin) [inline sonar-project.properties:4]: Could we remove everything here except for the top two lines please, want to keep it simple, less maintenance needed for these files then, twincam is on version 0.7, want to keep that number in a single place.

File: tests/sonar.sh [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+    git clean -fdx > /dev/null 2>&1
+  fi
+}
+
+# sonarcloud
+export SONAR_TOKEN="56af817d0e1cab1e8ba71fd26bfee5d397f845dd"
+
+projectKey= sed '1q;d' sonar-project.properties #sonar-project.properties
+projectOrg= sed '2q;d' sonar-project.properties #sonar-project.properties
+projectName= sed '5q;d' sonar-project.properties #sonar-project.properties
```
Comment (ericcurtin) [inline tests/sonar.sh:19]: Could we remove these three lines if we don't need them please

File: tests/sonar.sh [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+mkdir -p $HOME/.sonar
+SONAR_SCANNER_VERSION="4.7.0.2747"
+SONAR_SCANNER_DOWNLOAD_URL="https://binaries.sonarsource.com/Distribution/sonar-scanner-cli/sonar-scanner-cli-$SONAR_SCANNER_VERSION-linux.zip"
+curl -sSLo $HOME/.sonar/sonar-scanner.zip $SONAR_SCANNER_DOWNLOAD_URL
+unzip -o $HOME/.sonar/sonar-scanner.zip -d $HOME/.sonar/
+echo $projectname >>$HOME/.sonar/sonar-scanner-$SONAR_SCANNER_VERSION-linux/conf/sonar-scanner.properties
+echo $projectKey >>$HOME/.sonar/sonar-scanner-$SONAR_SCANNER_VERSION-linux/conf/sonar-scanner.properties
+echo $projectOrg >>$HOME/.sonar/sonar-scanner-$SONAR_SCANNER_VERSION-linux/conf/sonar-scanner.properties
```
Comment (ericcurtin) [inline tests/sonar.sh:27]: Same here

**Review by ericcurtin — COMMENTED** [review]
LGTM just some minor edits needed don't worry about any hard-coded strings, I'll get to them.

**Comment by iamianmullins** [comment]
Thanks @ericcurtin,  I think I've addressed all edits mentioned above. Let me know if there is anything else

**Comment by ericcurtin** [comment]
Squashed and merged via CLI

**Pull request closed without merging**